src/main/java/org/acra/util/TlsSniSocketFactory.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * This class was copied from this blog post:
+ * http://blog.dev001.net/post/67082904181/android-using-sni-and-tlsv1-2-with-apache-httpclient
+ * Thanks go to Dev001!
+ */
+package org.acra.util;
+
+import android.annotation.TargetApi;
+import android.net.SSLCertificateSocketFactory;
+import android.os.Build;
+import android.util.Log;
+
+import org.apache.http.conn.scheme.LayeredSocketFactory;
+import org.apache.http.conn.ssl.StrictHostnameVerifier;
+import org.apache.http.params.HttpParams;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.Socket;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.SSLPeerUnverifiedException;
+import javax.net.ssl.SSLSession;
+import javax.net.ssl.SSLSocket;
+
+@TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
```

Comment:
SSLCertificateSocketFactory.setHostname() requires at least Jelly Bean, else a fallback is used. I've moved the annotation to the method that is affected by this.